File: python/src/worker.py
import asyncio

from temporalio.client import Client
from temporalio.envconfig import ClientConfig
from temporalio.worker import Worker

from activities import (
    book_flight,
    book_hotel,
    book_car,
    notify_user,
    undo_book_flight,
    undo_book_hotel,
    undo_book_car,
)
from workflow import BookWorkflow


async def main():
    connect_config = ClientConfig.load_client_connect_config()
    client = await Client.connect(**connect_config)
    print(f"✅ Client connected to {client.service_client.config.target_host} in namespace '{client.namespace}'")

    worker = Worker(
        client,
        task_queue="trip-task-queue",
        workflows=[BookWorkflow],
        activities=[
            book_flight,
            book_hotel,
            book_car,
            undo_book_flight,
            undo_book_hotel,
            undo_book_car,
            notify_user,
        ],
    )

    print("Python trip booking worker starting...")
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: ruby/src/worker.rb
# frozen_string_literal: true

require_relative 'activities'
require_relative 'workflow'
require 'logger'
require 'temporalio/client'
require 'temporalio/env_config'
require 'temporalio/worker'

puts "⚙️ Using TEMPORAL_PROFILE: '#{ENV['TEMPORAL_PROFILE']}'"
args, kwargs = Temporalio::EnvConfig::ClientConfig.load_client_connect_options
kwargs[:logger] = Logger.new($stdout, level: Logger::INFO)
client = Temporalio::Client.connect(*args, **kwargs)
puts "✅ Client connected to '#{args[0]}' in namespace '#{args[1]}'"

worker = Temporalio::Worker.new(
  client:,
  task_queue: 'trip-task-queue',
  workflows: [
    TripBooking::BookWorkflow
  ],
  activities: [
    TripBooking::Activities::BookFlight,
    TripBooking::Activities::BookHotel,
    TripBooking::Activities::BookCar,
    TripBooking::Activities::NotifyUser,
    TripBooking::Activities::UndoBookFlight,
    TripBooking::Activities::UndoBookHotel,
    TripBooking::Activities::UndoBookCar
  ]
)

# Run the worker until SIGINT. This can be done in many ways, see "Workers" section for details.
puts 'Starting Worker (press Ctrl+C to exit)'
worker.run(shutdown_signals: ['SIGINT'])
